### PR TITLE
[WIP] return Colliding Bodies' Names if user given state is in collision

### PR DESCRIPTION
**Pull request by aditya-vk**
This PR forces the planning procedure to terminate if the user provided start state is in collision and prints the body nodes that are in collision.

WIP: Needs tests.
***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [ ] Set version target by selecting a milestone on the right side
- [ ] Summarize this change in `CHANGELOG.md`
- [ ] Add unit test(s) for this change

### CODE REVIEW

**Comment by aditya-vk** [comment]
Great suggestions! 
Yeah, this is quite messy but was a quick hack-around to unblock some of the planning issues we were facing with `pantry-loading` demo (planner returns an `invalid start state` due to fingers colliding with each other when the hand is closed but we had no access to this information previously). Created a PR so this issue can be tracked and completed appropriately soon. I should've probably left it as WIP.

File: include/aikido/planner/ompl/detail/Planner-impl.hpp [hunk 1]
```diff
@@ -24,6 +24,11 @@ trajectory::InterpolatedPtr planOMPL(
     double _maxPlanTime,
     double _maxDistanceBtwValidityChecks)
 {
+
```
Comment (jslee02) [inline include/aikido/planner/ompl/detail/Planner-impl.hpp]: Nit: Please remove unnecessary this whitespace

File: include/aikido/planner/ompl/detail/Planner-impl.hpp [hunk 2]
```diff
@@ -73,6 +78,10 @@ trajectory::InterpolatedPtr planOMPL(
...
```
Comment (jslee02) [inline include/aikido/planner/ompl/detail/Planner-impl.hpp]: Nit: Please remove unnecessary this whitespace

File: src/planner/SnapPlanner.cpp [hunk 3]
```diff
@@ -17,6 +17,10 @@ trajectory::InterpolatedPtr planSnap(
     const std::shared_ptr<aikido::constraint::Testable>& constraint,
     aikido::planner::PlanningResult& planningResult)
 {
+
```
Comment (jslee02) [inline src/planner/SnapPlanner.cpp]: Nit: Please remove unnecessary this whitespace

File: src/planner/ompl/Planner.cpp [hunk 4]
```diff
@@ -218,6 +218,10 @@ trajectory::InterpolatedPtr planCRRT(
     double _maxDistanceBtwProjections,
     double _minStepsize)
 {
+
```
Comment (jslee02) [inline src/planner/ompl/Planner.cpp]: Nit: Please remove unnecessary this whitespace

File: src/planner/ompl/Planner.cpp [hunk 5]
```diff
@@ -303,6 +307,10 @@ trajectory::InterpolatedPtr planCRRTConnect(
     double _minStepsize,
     double _minTreeConnectionDistance)
 {
+
```
Comment (jslee02) [inline src/planner/ompl/Planner.cpp]: Nit: Please remove unnecessary this whitespace

File: include/aikido/constraint/Testable.hpp [hunk 6]
```diff
@@ -14,8 +14,10 @@ class Testable
   virtual ~Testable() = default;
 
   /// Returns true if state satisfies this constraint.
+  /// \param _givenState: true if state is user-provided
```
[the remark this answers is not in the document]
Comment (jslee02) [reply include/aikido/constraint/Testable.hpp]: Nit: We don't use `:` (semicolon) for a parameter docstring.

[the remark this answers is not in the document]
Comment (jslee02) [reply include/aikido/constraint/Testable.hpp]: I guess `_state` is set to `false` when the checking state is provided by a planner, and, in that case, we don't want to terminate the planning due (could reject the sample though).

I think we don't need `_givenState` parameter for enforcing the planner to terminate. It's actually done by the caller (see [this](https://github.com/personalrobotics/aikido/pull/235/files#diff-af695089f1b391078609c33041298291R29)). `_givenState` is just using to determine whether a concrete `Testable` prints.

I agree with @mkoval that changing the way of receiving debug information.

**Comment by brianhou** [comment]
I think we'll wind up rewriting this entirely per @mkoval's suggestion, so I'm just going to close this PR. The branch will still be somewhat useful until we have the new implementation, though.

**Pull request closed without merging**